app/models/course/forum.rb

```diff
@@ -4,8 +4,62 @@ class Course::Forum < ActiveRecord::Base
 
   belongs_to :course, inverse_of: :forums
   has_many :topics, dependent: :destroy, inverse_of: :forum
+  has_many :posts, class_name: Course::Discussion::Post.name, through: :topics
   has_many :subscriptions, dependent: :destroy, inverse_of: :forum
 
+  # @!method self.with_topic_count
+  #   Augments all returned records with the number of topics in that forum.
+  scope :with_topic_count, (lambda do
+    joins { topics.outer }.
+      select { 'course_forums.*' }.
+      select { count(topics.id).as(topic_count) }.
+      group { course_forums.id }
+  end)
+
+  # @!method self.with_topic_post_count
+  #   Augments all returned records with the number of topic posts in that forum.
+  scope :with_topic_post_count, (lambda do
+    joins { posts.outer }.
+      select { 'course_forums.*' }.
+      select { count(posts.id).as(topic_post_count) }.
+      group { course_forums.id }
+  end)
+
+  # @!method self.with_topic_view_count
+  #   Augments all returned records with the number of topic views in that forum.
+  scope :with_topic_view_count, (lambda do
+    joins { topics.outer.views.outer }.
+      select { 'course_forums.*' }.
+      select { count(course_forum_topic_views.id).as(topic_view_count) }.
+      group { course_forums.id }
+  end)
+
+  # @!method self.with_all_count
+  #   Augments all returned records with the number of topics, topic posts and topic views
+  #   in that forum.
+  scope :with_all_count, (lambda do
```

Comment:
if the relation gives a syntax error then this is fine. but call it `with_forum_statistics`